config/internal/base/deployment.yaml.tmpl

```diff
@@ -59,10 +59,15 @@ spec:
           env:
             - name: MM_SERVICE_NAME
               value: {{.ServiceName}}
-              # External gRPC port of the service, should match ports.containerPort
+            # External endpoint for a payload processing service
+            {{if .PayloadProcessors}}
             - name: MM_PAYLOAD_PROCESSORS
               value: {{.PayloadProcessors}}
-              # External endpoint for a payload processing service
+            {{- else }}
+            - name: MM_PAYLOAD_PROCESSORS
+              value:
+            {{end}}
```

Comment:
Actually the reason I was suggesting this change was to avoid this... so that the env var is not set at all when no payload processors are configured. Could we remove the else block?

Reply:
So I tried that, but removing the else block doing that breaks the entire config; the test suite complains that it cannot find any of the yaml fields after the {{end}}

Reply:
Here are the results of `make run test`, keeping everything else identical, but just deleting the else block:

[error.log](https://github.com/kserve/modelmesh-serving/files/10981578/error.log)

Reply:
@RobGeada might it be something to do with the fact that `.PayloadProcessors` isn't a boolean and so a slightly different `{{if ..` is needed? Or that the .golden file needs to be updated to remove the env var name in the absent cases?

Failing that, maybe would be better to just add it to `[addMMEnvVars](https://github.com/kserve/modelmesh-serving/blob/main/controllers/modelmesh/modelmesh.go#L215)` instead as suggested in prior comment?

Reply:
Okay @njhill, done